### ISSUE TITLE
IndexError after restart on witti

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-beacon-chain#1153]
My node wasn't running for quite some while. Starting it again gave this error.  100% reproducable.

```
DBG 2020-06-09 15:31:11+02:00 Querying block number                      tid=144201 file=mainchain_monitor.nim:497 hash=099915f8fc1d2320795042fa631e349926e73f433977d214d084004a2aa7158a
DBG 2020-06-09 15:31:11+02:00 Received blocks on request                 topics="syncman" tid=144201 file=sync_manager.nim:860 blocks_count=1 blocks_map=.x.. peer=16Uiu2HAkvfDTKtFoSU5CHoA47BRwcJhN2zfvmr1kZXAwbweV3pq2 peer_score=200 request_count=4 request_slot=60982 request_step=1
DBG 2020-06-09 15:31:11+02:00 Forward sync imported blocks               topics="beacnde" tid=144201 file=beacon_node.nim:467 count=1 local_head_slot=60982
DBG 2020-06-09 15:31:11+02:00 Block received                             tid=144201 file=beacon_node_common.nim:100 blockRoot=ea46549e cat=block_listener pcs=receive_block signedBlock="(slot: 60983, proposer_index: 485, parent_root: \"c6567519\", state_root: \"97318553\", proposer_slashings_len: 0, attester_slashings_len: 0, attestations_len: 9, deposits_len: 0, voluntary_exits_len: 0)"
DBG 2020-06-09 15:31:11+02:00 Unresolved block (parent missing)          topics="clearblk" tid=144201 file=clearance.nim:240 blck="(slot: 60983, proposer_index: 485, parent_root: \"c6567519\", state_root: \"97318553\", proposer_slashings_len: 0, attester_slashings_len: 0, attestations_len: 9, deposits_len: 0, voluntary_exits_len: 0)" blockRoot=ea46549e cat=filtering missing=1 pcs=block_addition pending=1
DBG 2020-06-09 15:31:11+02:00 Block pool rejected peer's response        topics="beacnde" tid=144201 file=sync_manager.nim:516 blocks_count=1 blocks_map=.x.. errCode=MissingParent peer=16Uiu2HAkvfDTKtFoSU5CHoA47BRwcJhN2zfvmr1kZXAwbweV3pq2 request_count=4 request_slot=60982 request_step=1
 peers: 0 ❯ finalized: ffef3a02:1903 ❯ head: daa529d9:1905:22 ❯ time: 3051:23 (97655)                                                                                                                                                                                          ETH: 0 Traceback (most recent call last, using override)
/home/deme/repos/nim-beacon-chain/beacon_chain/mainchain_monitor.nim(516) main
/home/deme/repos/nim-beacon-chain/beacon_chain/mainchain_monitor.nim(509) NimMain
/home/deme/repos/nim-beacon-chain/beacon_chain/beacon_node.nim(1013) main
/home/deme/repos/nim-beacon-chain/beacon_chain/beacon_node.nim(763) start
/home/deme/repos/nim-beacon-chain/vendor/nim-chronicles/chronicles.nim(331) run
/home/deme/repos/nim-beacon-chain/vendor/nimbus-build-system/vendor/Nim/lib/system/excpt.nim(407) reportUnhandledError
/home/deme/repos/nim-beacon-chain/vendor/nimbus-build-system/vendor/Nim/lib/system/excpt.nim(358) reportUnhandledErrorAux
Error: unhandled exception: index out of bounds, the container is empty [IndexError]
```

Full log: https://gist.github.com/kdeme/53fb5f86c9cd068b762533d2a1b85982

**Comment by kdeme** [comment]
Added some extra logs, seems to die on this check: https://github.com/status-im/nim-beacon-chain/blob/devel/beacon_chain/sync_manager.nim#L532

Adding `debug "Suspects length", len=sq.suspects.len` there gives:
```
DBG 2020-06-09 15:49:11+02:00 Block pool rejected peer's response        topics="beacnde" tid=145007 file=sync_manager.nim:516 blocks_count=1 blocks_map=.x.. errCode=MissingParent peer=16Uiu2HAkvfDTKtFoSU5CHoA47BRwcJhN2zfvmr1kZXAwbweV3pq2 request_count=4 request_slot=60982 request_step=1
DBG 2020-06-09 15:49:11+02:00 Suspects length                            topics="beacnde" tid=145007 file=sync_manager.nim:532 len=0

```

**Comment by arnetheduck** [comment]
btw the easy way to get a proper callstack for index error is to `gdb` then `rbreak raiseIndex.*`

**Comment by cheatfate** [comment]
@arnetheduck the biggest problem here is not an `IndexError`. Database got corrupted somehow...

**Comment by cheatfate** [comment]
Fixed in #1160.
